compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -96,6 +96,27 @@ pub(crate) struct ProbeContext<'a, 'tcx> {
 
     scope_expr_id: HirId,
 
+    /// Delegation item can be expanded into method calls or fully qualified calls
+    /// depending on the callee's signature. Method calls are used to allow
+    /// autoref/autoderef for target expression. For example in:
+    ///
+    /// ```ignore (illustrative)
+    /// trait Trait : Sized {
+    ///     fn by_value(self) -> i32 { 1 }
+    ///     fn by_mut_ref(&mut self) -> i32 { 2 }
+    ///     fn by_ref(&self) -> i32 { 3 }
+    /// }
+    ///
+    /// struct NewType(SomeType);
+    /// impl Trait for NewType {
+    ///     reuse Trait::* { self.0 }
+    /// }
+    /// ```
+    ///
+    /// `self.0` will automatically coerce. The difference with existing method lookup
+    /// is that methods in delegation items are pre-resolved by callee path (`Trait::*`).
+    pre_resolved_method: Option<DefId>,
```

Comment:
```suggestion
    expected_def_id: Option<DefId>,
```
(In other places too.)